source/material_model/reactive_fluid_transport.cc

```diff
@@ -211,84 +218,92 @@ namespace aspect
+                  fluid_out->permeabilities[q] = reference_permeability * std::pow(porosity,3) * std::pow(1.0-porosity,2);
 
-              fluid_out->fluid_densities[q] = reference_rho_f * std::exp(fluid_compressibility * (in.pressure[q] - this->get_surface_pressure()));
+                  fluid_out->fluid_densities[q] = reference_rho_f * std::exp(fluid_compressibility * (in.pressure[q] - this->get_surface_pressure()));
 
-              if (in.requests_property(MaterialProperties::viscosity))
-                {
-                  const double phi_0 = 0.05;
+                  if (in.requests_property(MaterialProperties::viscosity))
+                    {
+                      const double phi_0 = 0.05;
 
-                  // Limit the porosity to be no smaller than 1e-8 when
-                  // calculating fluid effects on viscosities.
-                  porosity = std::max(porosity,1e-8);
-                  fluid_out->compaction_viscosities[q] = std::max(std::min(out.viscosities[q] * shear_to_bulk_viscosity_ratio * phi_0/porosity, max_compaction_visc), min_compaction_visc);
+                      // Limit the porosity to be no smaller than 1e-8 when
+                      // calculating fluid effects on viscosities.
+                      porosity = std::max(porosity,1e-8);
+                      fluid_out->compaction_viscosities[q] = std::max(std::min(out.viscosities[q] * shear_to_bulk_viscosity_ratio * phi_0/porosity, max_compaction_visc), min_compaction_visc);
+                    }
                 }
             }
-        }
 
-      ReactionRateOutputs<dim> *reaction_rate_out = out.template get_additional_output<ReactionRateOutputs<dim>>();
+          ReactionRateOutputs<dim> *reaction_rate_out = out.template get_additional_output<ReactionRateOutputs<dim>>();
 
-      // Fill reaction rate outputs if the model uses operator splitting.
-      // Specifically, change the porosity (representing the amount of free water)
-      // based on the water solubility and the water content.
-      if (this->get_parameters().use_operator_splitting && reaction_rate_out != nullptr)
-        {
-          std::vector<double> eq_free_fluid_fractions(out.n_evaluation_points());
-          melt_fractions(in, eq_free_fluid_fractions);
-
-          for (unsigned int q=0; q<out.n_evaluation_points(); ++q)
-            for (unsigned int c=0; c<in.composition[q].size(); ++c)
-              {
-                double porosity_change = eq_free_fluid_fractions[q] - in.composition[q][porosity_idx];
-                // do not allow negative porosity
-                if (in.composition[q][porosity_idx] + porosity_change < 0)
-                  porosity_change = -in.composition[q][porosity_idx];
+          // Fill reaction rate outputs if the model uses operator splitting.
+          // Specifically, change the porosity (representing the amount of free water)
+          // based on the water solubility and the water content.
```

Comment:
```suggestion
          // based on the fluid solubility and the fluid content.
```

Reply:
This and the above comment are regarding code not introduced in this PR, but I think it would be worth changing it now if I indeed this part of the code is not specific to one of the water-based reaction models (it looks like this is not the case).